azure-quantum/azure/quantum/qiskit/provider.py

```diff
@@ -132,11 +116,25 @@ def backends(self, name=None, **kwargs):
         backend_list = [x for v in self._backends.values() for x in v]
 
         # filter by properties specified in the kwargs and filter function
-        backends: List[Backend] = self._filter_backends(
+        filtered_backends: List[Backend] = self._filter_backends(
             backend_list, filters=workspace_allowed, **kwargs
         )
 
-        return backends
+        # Also filter out non-default backends.
+        default_backends = list(
+            filter(
+                lambda backend: self._match_all(
+                    backend.configuration().to_dict(), {"is_default": True}
+                ),
+                filtered_backends,
+            )
+        ) 
+        # In some cases, for example due to certain filters, there may be no default backends, 
+        # in this case return previous version of the list.  
```

Comment:
nit: The comment contradicts with the code below it. May be say something like:
If default backends were found - return them, otherwise return the `filtered_backends` collection. The latter case could happen where there's no default backend defined for the specified target.